### ISSUE TITLE
Socket leak in Queue.ReceiveOne

### ISSUE THREAD

**Issue by g-rad** [Azure/azure-service-bus-go#72]
I've been playing with Queue.Receive and Queue.ReceiveOne and noticed that ReceiveOne creates a new receiver every time. Below example of code with socket leak:
```
q,_ := NewQueueClient("sb-name", "key-name", "name", "queue-name")

for {
	ctx,_ := context.WithTimeout(context.Background(), time.Second*5)

	err := q.ReceiveOne(ctx, servicebus.HandlerFunc(func(ctx context.Context, msg *servicebus.Message) servicebus.DispositionAction {
		return msg.Complete()
	}))
	log.Println("Error:", err)
}
```

The obvious workaround would be to call `queue.Close()` after every iteration but I believe `Queue.ReceiveOne()` should know how to handle existing receivers.

The reason I'm using `Queue.ReceiveOne()` instead of `Queue.Receive()` is last time I used it ReceiveOne didn't receive several last messages out of hundred. I'm still debugging it.

**Comment by g-rad** [comment]
@devigned I added logging to `Namespace.newConnection()` and can clearly see it's called for every `ReceiveOne` of my original code. So I believe a new connection is created each time.

`netstat -an | grep ESTABLISHED | wc -l` also shows increasing number of opened sockets.

**Comment by devigned** [comment]
Yep... that is for sure a bug. Looking a bit closer at `ensureReceiver`, it is not checking if there is an existing receiver. Thank you!
